controller_manager/test/test_controller_manager.cpp

```diff
@@ -495,7 +495,7 @@ TEST_P(TestControllerUpdateRates, check_the_controller_update_rate)
       // cycles it will have 369 instead of 370.
       EXPECT_NEAR(
         test_controller->internal_counter - initial_counter,
-        (controller_update_rate * no_of_secs_passed), 1);
+        (static_cast<double>(controller_update_rate) * no_of_secs_passed), 1.0);
```

Comment:
May be the  `EXPECT_NEAR` here excepts the double type here?

Reply:
yes, this was a thing. Maybe we should simply add a bit more complex logic to allow the multiplication to be smaller then `something + 1`.

Or what do you want to test here exaclty?

Reply:
Ideally, they have to be equal, but for the initial cycle set and while configuring the controller and all, we lose one for some reason, in the case of non-perfect divisors. However, we can proceed with this change. I don't think it's going to make a big difference

Reply:
we can be more explicit here fellas

```suggestion
 EXPECT_THAT(
        test_controller->internal_counter - initial_counter,
        testing::Or(Eqcontroller_update_rate* no_of_secs_passed, (controller_update_rate * no_of_secs_passed)+1));
```

I believe something like this is also supported

Reply:
@bmagyar Thanks. I learned something today. 

If we use this, then we need to do a small change `Eq((controller_update_rate * no_of_secs_passed)+1` to `Eq((controller_update_rate * no_of_secs_passed)-1`.

Reply:
`testing::AnyOf` is the thing!